src/dynsight/_internal/vision/detect.py

```diff
@@ -51,115 +59,127 @@ def __init__(
         Creates all necessary subdirectories and extracts video frames if not
         already present.
 
-        Args:
-            input_video: The input video object from which to extract frames.
-            project_folder: Path to the main project directory.
+        Parameters:
+            input_video:
+                The input video object from which to extract frames.
 
+            project_folder:
+                Path to the main project directory.
         """
-        # Define the project folder path
-        self.project_folder = project_folder
+        self._project_folder = project_folder
+        """Main directory for all the project outputs."""
```

Comment:
Unsure why these are here, please bring back the attributes list?

Reply:
Ahh, I see, because they are private now - the solution here is to use comments, not `"""Text"""` here, and have a `get_XX(): return self._XX` method for anything you think the users will want access to.